kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -716,10 +717,20 @@ functionAnd first second
     & pure
   | otherwise = empty
   where
-    (first', second')
-      | isConstructorLike first = (first, second)
-      | isConstructorLike second = (second, first)
-      | otherwise = minMax first second
+    (first', second') =
+        case compareForEquals first second of
+            LT -> (first, second)
+            _  -> (second, first)
+
+compareForEquals
```

Comment:
Please add documentation for this function, in particular, what does "for equals" mean in this context?